### PR TITLE
[syncd] Add pre match logic for acl entry

### PR DESCRIPTION
**Pull request by kcudnik**
this will test scenario where we have 1 acl entry with counter and after
warm boot 2 acl entries and 2 counters the same which could end up
matching wrong acl counter and causing fail "object exist" on broadcom
acl entry set, which will require improve pre match logic for acl entry

### CODE REVIEW

File: syncd/ComparisonLogic.cpp
Comment (vaibhavhd) [inline syncd/ComparisonLogic.cpp:2834]: Rookie question: are ACL priorities guaranteed to be always unique? If not, then we will likely run into same issue again.

File: syncd/ComparisonLogic.cpp
Comment (kcudnik) [reply syncd/ComparisonLogic.cpp:3098]: no, since ACL_ENTRY is leaf object

File: syncd/ComparisonLogic.cpp
Comment (kcudnik) [reply syncd/ComparisonLogic.cpp:2834]: no they are not guaranteed in general, but sonic make sure that they are, since we want to control the right order of the acl, so having 2 acls with the same priority not guarantee the order of them

**Review by vaibhavhd — DISMISSED** [review]
Please wait for Ying or Sai's review before merge.

File: syncd/ComparisonLogic.cpp
Comment (saiarcot895) [inline syncd/ComparisonLogic.cpp:2980]: Should this be before the log message above, so that the right prematch count gets printed?

Comment (kcudnik) [reply syncd/ComparisonLogic.cpp:2980]: yes

Comment (kcudnik) [reply syncd/ComparisonLogic.cpp]: fixed

**Review by saiarcot895 — APPROVED** [review]

**Comment by qiluo-msft** [comment]
This commit could not be cleanly cherry-picked to 202012. Please submit another PR.

**Comment by vaibhavhd** [comment]
@kcudnik please raise a separate PR for 202012 branch

**Comment by liuh-80** [comment]
Manually cherry-pick PR created: https://github.com/sonic-net/sonic-sairedis/pull/1257

**Pull request merged**
